### PR TITLE
Add test to verify failure for renaming to long table name

### LINKED ISSUE #13073: Add test for ALTER TABLE ... RENAME TO long identifier
Relates to #12882 & #12892

### PR DESCRIPTION
**Pull request by ebyhr**
## Description

Add test to verify failure for renaming to long table name. The changes for MongoDB and SQL Server connectors are required to pass the test.
Fixes #13073

## Documentation

(x) No documentation is needed.

## Release notes

(x) Release notes entries required with the following suggested text:

```markdown
# MongoDB, SQL Server, PostgreSQL
* Prevent renaming to a table with a name longer than the max length.
  Previously, the name was truncated to the max length. ({issue}`issuenumber`)
```

### CODE REVIEW

**Comment by ebyhr** [comment]
CI `plugin/trino-redis` failed by
```
Status 500: {"message":"Head \"[https://registry-1.docker.io/v2/testcontainers/ryuk/manifests/0.3.3\](https://registry-1.docker.io/v2/testcontainers/ryuk/manifests/0.3.3/)": Get \"https://auth.docker.io/token?account=githubactions&scope=repository%3Atestcontainers%2Fryuk%3Apull&service=registry.docker.io\": EOF"}
```

File: testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java [hunk 1]
```diff
@@ -2011,12 +2011,8 @@ public void testCreateTableWithLongTableName()
         }
 
         String invalidTableName = validTableName + "z";
-        try {
-            assertUpdate("CREATE TABLE " + invalidTableName + " (a bigint)");
-        }
-        catch (Throwable e) {
-            verifyTableNameLengthFailurePermissible(e);
-        }
+        assertThatThrownBy(() -> assertUpdate("CREATE TABLE " + invalidTableName + " (a bigint)"))
```
Comment (hashhar) [inline testing/trino-testing/src/main/java/io/trino/testing/BaseConnectorTest.java:2013]: Long commit message - ` Ensure failure when creating invalid table name in testCreateTableWithLongTableName`.

Would `Verify failure for invalid table name in testCreateTableWithLongTableName` work?

File: plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java [hunk 2]
```diff
@@ -474,14 +473,13 @@ public void dropTable(ConnectorSession session, JdbcTableHandle handle)
...
-    protected void renameTable(ConnectorSession session, String catalogName, String schemaName, String tableName, SchemaTableName newTable)
+    protected String renameTableSql(String catalogName, String remoteSchemaName, String remoteTableName, String newRemoteSchemaName, String newRemoteTableName)
     {
-        String sql = format("RENAME TABLE %s.%s TO %s.%s",
-                quoted(schemaName),
-                quoted(tableName),
-                quoted(newTable.getSchemaName()),
-                quoted(newTable.getTableName()));
-        execute(session, sql);
+        return format("RENAME TABLE %s.%s TO %s.%s",
```
Comment (hashhar) [inline plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/ClickHouseClient.java:478]: Nice, also seems to fix the issue that we would throw raw SQLException instead of wrapping into a TrinoException.

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java [hunk 3]
```diff
@@ -67,6 +68,8 @@
 {
     private static final Logger log = Logger.get(MongoMetadata.class);
 
+    private static final int MAX_QUALIFIED_IDENTIFIER_LENGTH = 120;
```
Comment (hashhar) [inline plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java:71]: nit: ` Check target table name when renaming a table in MongoDB ` -> ` Check target table name length when renaming a table in MongoDB `

Comment (hashhar) [reply plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java:71]: The MongoDB docs talk about bytes and it seems they don't disallow multibyte characters so instead of checking string length we should check bytes probably.

File: plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java [hunk 4]
```diff
@@ -277,6 +278,16 @@ protected boolean isTableLockNeeded(ConnectorSession session)
         return isBulkCopyForWrite(session) && isBulkCopyForWriteLockDestinationTable(session);
     }
 
+    @Override
+    protected void verifyTableName(DatabaseMetaData databaseMetadata, String tableName)
+            throws SQLException
+    {
+        // SQL Server truncates table name to 128 chars silently when renaming a table
```
Comment (hashhar) [inline plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java:285]: nit: The comment is slightly confusing because it talks about 128 but does comparison with `databaseMetadata.getMaxTableNameLength()`.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java [hunk 5]
```diff
@@ -656,6 +657,18 @@ protected String tableDefinitionForQueryLoggingCount()
                 ")";
     }
 
+    @Override
+    protected OptionalInt maxTableNameLength()
+    {
+        return OptionalInt.of(256);
```
Comment (hashhar) [inline plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:663]: I don't see explicit checks in KuduMetadata yet. Is the goal of this change only to enforce limit on our side for connectors that we know will silently truncate table names? Or do we want to always enforce the limits?

Comment (hashhar) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:663]: No change requested - just a question to understand scope of change.

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java (hunk 3 above)
Comment (ebyhr) [reply plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java:71]: Good catch. Let me fix.

File: plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java (hunk 5 above)
Comment (ebyhr) [reply plugin/trino-kudu/src/test/java/io/trino/plugin/kudu/TestKuduConnectorTest.java:663]: The current goal is just ensuring the failure and the place (remote or connector) is out of scope.

File: plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java (hunk 3 above)
Comment (hashhar) [reply plugin/trino-mongodb/src/main/java/io/trino/plugin/mongodb/MongoMetadata.java:72]: nit: the var should probably be renamed since it's bytes now instead of length.

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java [hunk 6]
```diff
@@ -5485,7 +5484,8 @@ protected OptionalInt maxTableNameLength()
     {
         // This value depends on metastore type
         // The connector appends uuids to the end of all table names
-        return OptionalInt.of(255 - UUID.randomUUID().toString().length());
+        // 33 is the length of random suffix. e.g. {table name}-142763c594d54e4b9329a98f90528caf
+        return OptionalInt.of(255 - 33);
```
Comment (hashhar) [inline plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java:5513]: why this change?

**Review by hashhar — APPROVED** [review]

File: plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java (hunk 6 above)
Comment (ebyhr) [reply plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorTest.java:5513]: Previously, this test passed in Iceberg unintentionally because of it's try-catch style. This change is required to adjust to the right max length.

**Review by hashhar — APPROVED** [review]

**Comment by ebyhr** [comment]
CI hit #13107
